.github/workflows/Linux.yml

```diff
@@ -122,94 +115,96 @@ jobs:
         [ODBC]
         Trace     = 1
         TraceFile = ${{ github.workspace }}/run/odbc-driver-manager-trace.log
         Debug     = 1
         DebugFile = ${{ github.workspace }}/run/odbc-driver-manager-debug.log
-
+        
         [ODBC Data Sources]
         ClickHouse DSN (ANSI)         = ClickHouse ODBC Driver (ANSI)
         ClickHouse DSN (Unicode)      = ClickHouse ODBC Driver (Unicode)
         ClickHouse DSN (ANSI, RBWNAT) = ClickHouse ODBC Driver (ANSI)
-
+        
         [ClickHouse DSN (ANSI)]
         Driver        = ClickHouse ODBC Driver (ANSI)
         Description   = Test DSN for ClickHouse ODBC Driver (ANSI)
         Url           = http://${CLICKHOUSE_SERVER_IP}
         DriverLog     = yes
         DriverLogFile = ${{ github.workspace }}/run/clickhouse-odbc-driver.log
-
+        
         [ClickHouse DSN (Unicode)]
         Driver        = ClickHouse ODBC Driver (Unicode)
         Description   = Test DSN for ClickHouse ODBC Driver (Unicode)
-        Url           = http://${CLICKHOUSE_SERVER_IP}
+        Url           = http://localhost:8123
         DriverLog     = yes
         DriverLogFile = ${{ github.workspace }}/run/clickhouse-odbc-driver-w.log
-
+        
         [ClickHouse DSN (ANSI, RBWNAT)]
         Driver        = ClickHouse ODBC Driver (ANSI)
         Description   = Test DSN for ClickHouse ODBC Driver (ANSI) that uses RowBinaryWithNamesAndTypes as data source communication default format
-        Url           = http://${CLICKHOUSE_SERVER_IP}/query?default_format=RowBinaryWithNamesAndTypes
+        Url           = http://localhost:8123/query?default_format=RowBinaryWithNamesAndTypes
         DriverLog     = yes
         DriverLogFile = ${{ github.workspace }}/run/clickhouse-odbc-driver.log
         EOF
 
+    # Run all tests except those that were run in "Test - unit tests" step.
+    - name: Test - Run C++ integration tests
+      working-directory: ${{ github.workspace }}/build
+      run: |
+        export ODBCSYSINI=
+        export ODBCINSTINI="${{ github.workspace }}/run/.odbcinst.ini"
+        export ODBCINI="${{ github.workspace }}/run/.odbc.ini"
         if [[ "${{ matrix.odbc_provider }}" == "iODBC" ]]; then
-            export GTEST_FILTER="-PerformanceTest.*"
+          export GTEST_FILTER="-PerformanceTest.*"
         fi
-
-        # Run all tests except those that were run in "Test - unit tests" step.
         ctest --output-on-failure --build-config ${{ matrix.build_type }} -E '.*-ut.*'
+
+    - name: Prepare Python dependencies
+      working-directory: source/test
+      run: |
+        pip install -r requirements.txt
+
+      # An empty `ODBCSYSINI` is required in this case to run properly with custom `ODBCINI`/`ODBCINSTINI` paths
+    - name: Test - Run Python e2e tests
+      working-directory: source/test
+      run: |
+        export ODBCSYSINI=
+        export ODBCINSTINI="${{ github.workspace }}/run/.odbcinst.ini"
+        export ODBCINI="${{ github.workspace }}/run/.odbc.ini"
```

Comment:
Probably should have been made a bit cleaner and extracted into some "generate env variables" step, but it works. Note the empty `ODBCSYSINI` (mentioned in the contribution guide now, too).